inst/include/lpdens.hpp

```diff
@@ -5,231 +5,286 @@
+            x_new = (x.array() - xmin_ + 5e-3) / (xmax_ - xmin_ + 1e-2);
+            x_new = stats::qnorm(x_new);
+        } else if (!std::isnan(xmin_)) {
+            // left boundary -> log transform
+            x_new = (1e-3 + x.array() - xmin_).log();
+        } else if (!std::isnan(xmax_)) {
+            // right boundary -> negative log transform
+            x_new = (1e-3 + xmax_ - x.array()).log();
         } else {
-            corr_term.fill(1.0);
+            // no boundary -> no transform
         }
-
-        return fhat.array() * corr_term.array();
-    }
-
-    Eigen::VectorXd construct_grid_points(const Eigen::VectorXd& x)
-    {
-        double x_min = x.minCoeff();
-        double x_max = x.maxCoeff();
-        double range = x_max - x_min;
-
-        size_t grid_size = 50;
-        Eigen::VectorXd lowr_ext, uppr_ext, grid_points(grid_size);
-
-        // no left boundary -> add a few points to the left
-        if (std::isnan(xmin_)) {
-            lowr_ext = Eigen::VectorXd::LinSpaced(5,
-                                                  x_min - 0.5 * range,
-                                                  x_min - 0.05 * range);
-            grid_size -= 5;
+    } else {
+        if (!std::isnan(xmin_) & !std::isnan(xmax_)) {
+            // two boundaries -> probit transform
+            x_new = stats::pnorm(x).array() + xmin_ - 5e-3;
+            x_new *=  (xmax_ - xmin_ + 1e-2);
+        } else if (!std::isnan(xmin_)) {
+            // left boundary -> log transform
+            x_new = x.array().exp() + xmin_ - 1e-3;
+        } else if (!std::isnan(xmax_)) {
+            // right boundary -> negative log transform
+            x_new = -(x.array().exp() - xmax_ - 1e-3);
         } else {
-            lowr_ext = Eigen::VectorXd();
+            // no boundary -> no transform
         }
+    }
 
-        // no right boundary -> add a few points to the right
-        if (std::isnan(xmax_)) {
-            uppr_ext = Eigen::VectorXd::LinSpaced(5,
-                                                  x_max + 0.05 * range,
-                                                  x_max + 0.5 * range);
-            grid_size -= 5;
-        } else {
-            uppr_ext = Eigen::VectorXd();
-        }
-
-        grid_points <<
-            lowr_ext,
-            Eigen::VectorXd::LinSpaced(grid_size, x_min, x_max),
-            uppr_ext;
-
-        return grid_points;
+    return x_new;
+}
+
+//! corrects the density estimate for a preceding boundary transformation of
+//! the data.
+//! @param x evaluation points (in original domain).
+//! @param fhat the density estimate evaluated in the transformed domain.
+//! @return corrected density estimates at `x`.
+inline Eigen::VectorXd LPDens1d::boundary_correct(const Eigen::VectorXd& x,
+                                                  const Eigen::VectorXd& fhat)
+{
+    Eigen::VectorXd corr_term(fhat.size());
+    if (!std::isnan(xmin_) & !std::isnan(xmax_)) {
```

Comment:
Didn't we have compatibility issues with the `std::isnan`? In `vinecopulib`, we do `(boost::math::isnan)(something)`.

Reply:
I do remember that, but don' remember why. I'd prefer to keep it simple as long there's no error.

Reply:
OK, let's keep this in mind when we do the continuous integration then.